pkg/chunk/cache/cache_test.go

```diff
@@ -82,12 +83,37 @@ func testCacheMultiple(t *testing.T, cache cache.Cache, keys []string, chunks []
 	require.Len(t, bufs, len(keys))
 	require.Len(t, missingKeys, 0)
 
-	foundChunks, missing, err := chunk.ProcessCacheResponse(chunks, found, bufs)
+	result := []chunk.Chunk{}
+	for i := range found {
+		c, err := chunk.ParseExternalKey(userID, found[i])
+		require.NoError(t, err)
+		err = c.Decode(chunk.NewDecodeContext(), bufs[i])
+		require.NoError(t, err)
+		result = append(result, c)
+	}
+	require.Equal(t, chunks, result)
+}
+
+func testChunkFetcher(t *testing.T, c cache.Cache, keys []string, chunks []chunk.Chunk) {
```

Comment:
The fact that the tests for ChunkFetcher live in a different package from the implementation is a bit weird to me. Perhaps some of these tests should live next to the fetcher? WDYT?

The behavior existed before this PR too, so I don't think it necessarily needs to be changed now.

Reply:
Yeah the challenge is circular imports; I've got a TODO item to complete reorganise the chunk package, as its a bit of a pain.